Drivers/ST7789-i8080/Source/St7789i8080Display.cpp

```diff
@@ -0,0 +1,342 @@
+#include "St7789i8080Display.h"
+#include <Tactility/Log.h>
+#include <driver/gpio.h>
+#include <esp_lcd_panel_io.h>
+#include <esp_lcd_panel_ops.h>
+#include <esp_lvgl_port.h>
+#include <freertos/FreeRTOS.h>
+#include <freertos/task.h>
+#include <lvgl.h>
+
+constexpr auto TAG = "St7789i8080Display";
+static St7789i8080Display* g_display_instance = nullptr;
+
+// ST7789 initialization commands
+typedef struct {
+    uint8_t cmd;
+    uint8_t data[14];
+    uint8_t len;
+} lcd_init_cmd_t;
+
+static const lcd_init_cmd_t st7789_init_cmds[] = {
+    {0x11, {0}, 0 | 0x80},
+    {0x36, {0x08}, 1},  
+    {0x3A, {0X05}, 1},
+    {0x20, {0}, 0},  
+    {0xB2, {0X0B, 0X0B, 0X00, 0X33, 0X33}, 5},
+    {0xB7, {0X75}, 1},
+    {0xBB, {0X28}, 1},
+    {0xC0, {0X2C}, 1},
+    {0xC2, {0X01}, 1},
+    {0xC3, {0X1F}, 1},
+    {0xC6, {0X13}, 1},
+    {0xD0, {0XA7}, 1},
+    {0xD0, {0XA4, 0XA1}, 2},
+    {0xD6, {0XA1}, 1},
+    {0xE0, {0XF0, 0X05, 0X0A, 0X06, 0X06, 0X03, 0X2B, 0X32, 0X43, 0X36, 0X11, 0X10, 0X2B, 0X32}, 14},
+    {0xE1, {0XF0, 0X08, 0X0C, 0X0B, 0X09, 0X24, 0X2B, 0X22, 0X43, 0X38, 0X15, 0X16, 0X2F, 0X37}, 14},
+};
+
+// Callback when color transfer is done
+static bool notify_lvgl_flush_ready(esp_lcd_panel_io_handle_t panel_io, 
+                                    esp_lcd_panel_io_event_data_t *edata, 
+                                    void *user_ctx) {
+    lv_display_t *disp = (lv_display_t *)user_ctx;
+    lv_display_flush_ready(disp);
+    return false;
+}
+
+St7789i8080Display::St7789i8080Display(const Configuration& config) 
+    : configuration(config), lock(std::make_shared<std::mutex>()) {
+    
+    // Validate configuration
+    if (!configuration.isValid()) {
+        TT_LOG_E(TAG, "Invalid configuration: resolution must be set");
+        return;
+    }
+    
+    // Calculate buffer size if needed
+    configuration.calculateBufferSize();
+    
+    // Allocate buffer based on resolution
+    size_t buffer_size = configuration.bufferSize * LV_COLOR_FORMAT_GET_SIZE(LV_COLOR_FORMAT_RGB565);
+    buf1 = (uint8_t*)heap_caps_malloc(buffer_size, MALLOC_CAP_DMA);
+    if (!buf1) {
+        TT_LOG_E(TAG, "Failed to allocate display buffer");
+    }
```

Comment:
Buffers should be allocated during `start()`.
As long as the driver isn't active, we it shouldn't take up much memory.